manifests/linux/redhat.pp

```diff
@@ -8,13 +8,24 @@
 #   Ensure parameter passed onto Service[] resources.
 #   Default: running
 #
+# [*ensure_v6*]
+#   Ensure parameter passed onto Service[] resources.
+#   Default: running
+#
 # [*enable*]
 #   Enable parameter passed onto Service[] resources.
 #   Default: true
 #
+# [*enable_v6*]
+#   Enable parameter passed onto Service[] resources.
+#   Default: true
+#
+#
 class firewall::linux::redhat (
   $ensure          = running,
+  $ensure_v6       = running,
   $enable          = true,
+  $enable_v6       = true,
```

Comment:
To remain backwards-compatible, `$ensure_v6` and `$enable_v6` have to default to the values of `$ensure` and `$enable`. Unfortunately this means the silly pattern of `undef` and then `pick()` to an underscore value: https://github.com/puppetlabs/puppetlabs-tomcat/blob/master/manifests/instance.pp#L49